sw/device/silicon_creator/lib/sigverify/BUILD

```diff
@@ -0,0 +1,220 @@
+        ],
+        host = [
+            "//sw/device/lib/base/testing:global_mock",
+            "//sw/device/silicon_creator/testing:mask_rom_test",
+            "@googletest//:gtest",
+        ],
+        shared = [
+            ":rsa_key",
+            "//sw/device/silicon_creator/lib:error",
+        ],
+    ),
+)
+
+opentitan_functest(
+    name = "mod_exp_otbn_functest",
+    srcs = ["mod_exp_otbn_functest.c"],
+    verilator = verilator_params(
+        timeout = "long",
+        tags = [
+            "manual",
+            "verilator",
+        ],
+    ),
+    deps = [
+        ":mod_exp_otbn",
+        ":testvectors",
+        "//sw/device/silicon_creator/lib:test_main",
+        "//sw/device/silicon_creator/lib/base:sec_mmio",
+    ],
+)
+
+# Common inputs for sigverify.c; we publish two variants of this
+# library, so factoring this out simplifies the build.
+cc_library(
+    name = "sigverify_inner",
+    srcs = ["sigverify.c"],
+    hdrs = ["sigverify.h"],
+    visibility = ["//visibility:private"],
+    deps = [
+        ":mod_exp_ibex",
+        ":mod_exp_otbn",
+        ":rsa_key",
+        "//hw/ip/otp_ctrl/data:otp_ctrl_regs",
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/lib/base:bitfield",
+        "//sw/device/lib/base:hardened",
+        "//sw/device/lib/base:macros",
+        "//sw/device/lib/base:memory",
+        "//sw/device/silicon_creator/lib:manifest",
+        "//sw/device/silicon_creator/lib/drivers:hmac",
+        "//sw/device/silicon_creator/lib/drivers:lifecycle",
+        "//sw/device/silicon_creator/lib/drivers:otp",
+    ],
+)
+
+cc_library(
+    name = "sigverify",
+    deps = [
+        ":mod_exp_ibex",
+        ":mod_exp_otbn",
+        ":sigverify_inner",
+    ],
+)
+
+cc_library(
+    name = "sigverify_without_mod_exp_ibex_mock",
+    testonly = True,
+    visibility = ["//sw/device/silicon_creator/mask_rom:__pkg__"],
+    deps = [
+        dual_cc_device_library_of(":mod_exp_ibex"),
+        ":mod_exp_otbn",
+        ":sigverify_inner",
+    ],
+)
+
+cc_test(
+    name = "sigverify_unittest",
+    srcs = ["sigverify_unittest.cc"],
+    deps = [
+        ":sigverify",
+        "//hw/ip/flash_ctrl/data:flash_ctrl_regs",
+        "//sw/device/silicon_creator/testing:mask_rom_test",
+        "@googletest//:gtest_main",
+    ],
+)
+
+opentitan_functest(
+    name = "sigverify_functest",
+    srcs = ["sigverify_functest.c"],
+    verilator = verilator_params(
+        timeout = "long",
+        tags = [
+            "manual",
+            "verilator",
+        ],
+    ),
+    deps = [
+        ":sigverify",
+        "//sw/device/silicon_creator/lib:test_main",
+        "//sw/device/silicon_creator/lib/base:sec_mmio",
+    ],
+)
+
+opentitan_functest(
+    name = "sigverify_dynamic_functest",
+    srcs = ["sigverify_dynamic_functest.c"],
+    verilator = verilator_params(
+        timeout = "long",
+        tags = [
+            "manual",
+            "verilator",
```

Comment:
you shouldn't need this as it is a required tag that gets automatically added [here](https://github.com/lowRISC/opentitan/blob/5852b71c5c0d70709699b4a18fc6de921bc34836/rules/opentitan_test.bzl#L140). You will need a `cpu:X` tag though **_if you want one_**, as that is not a required tag, [rather a default](https://github.com/lowRISC/opentitan/blob/5852b71c5c0d70709699b4a18fc6de921bc34836/rules/opentitan_test.bzl#L100).

Reply:
(this was my fault, I refactored the `verilator_params()` function and did not refactor the invocations to it; sorry about that. I will make a pass through the codebase for these once this merges)

Reply:
As discussed offline, let's do this as a separate cleanup in the name of maintaining velocity (i.e., CI being painfully slow).